libc/utils/MPFRWrapper/MPFRUtils.cpp

```diff
@@ -238,17 +242,24 @@ class MPFRNumber {
     mpfr_exp2m1(result.value, value, mpfr_rounding);
     return result;
 #else
-    MPFRNumber result(*this, mpfr_precision * 8);
-    mpfr_exp2(result.value, value, mpfr_rounding);
-
-    if (mpfr_underflow_p()) {
-      mpfr_clear_underflow();
-
-      if (mpfr_rounding == MPFR_RNDZ)
-        return underflow_rndz_fallback;
+    unsigned int prec = mpfr_precision * 3;
+    MPFRNumber result(*this, prec);
+
+    float f = mpfr_get_flt(abs().value, mpfr_rounding);
+    if (f > 0.5f && f < 0x1.0p30f) {
+      mpfr_exp2(result.value, value, mpfr_rounding);
+      mpfr_sub_ui(result.value, result.value, 1, mpfr_rounding);
+      return result;
```

Comment:
`underflow_rndz_fallback` is not needed anymore.  Remove it from the function signature.

Reply:
Oops.